marlowe-runtime/tx/Language/Marlowe/Runtime/Transaction/Constraints.hs

```diff
@@ -441,11 +442,12 @@ findMinUtxo protocol (chAddress, mbDatum, origValue) =
       -- FIXME Found what looks like the same value being added and then subtracted from this computation
       -- atLeastHalfAnAda = origValue <> C.lovelaceToValue (maximum [500_000, C.selectLovelace origValue] - C.selectLovelace origValue)
       atLeastHalfAnAda = C.lovelaceToValue (maximum [500_000, C.selectLovelace origValue])
+      revisedValue = origValue <> C.negateValue (C.lovelaceToValue $ C.selectLovelace origValue) <> atLeastHalfAnAda
```

Comment:
I see, so what was happening is only the ADA part of the value (or 500_000, if it's higher) was making it past this code. Now the calculation pulls the ADA out of `origValue`, and then puts a possibly-adjusted value back in.

Reply:
Correct. In particular, it was ignoring the non-Ada assets that use bytes and that therefore increase min-UTxO.